File: Selenium/Project/Activity5.java
package seleniumproject;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

public class Activity5 
{
	WebDriver driver;
	String message;
	WebDriverWait wait;
	Actions builder;
	
		
	@BeforeClass
	public void setup() 
    {
		//create a WebDriver instance
		driver = new FirefoxDriver();
		
		builder = new Actions(driver);
		
		//go to a web page
		driver.get("http://alchemy.hguy.co/orangehrm");

		//do stuff on the page
		System.out.println("Title of the page is : " + driver.getTitle());
    }
	
	@BeforeTest
	public void setupBeforeTest() 
    {
		// Initialize wait
        wait = new WebDriverWait(driver, Duration.ofSeconds(60));		
    }
 
	@AfterClass
	public void tearDown() 
	{
 	     // Close the driver
		driver.close();
	}
	
	@DataProvider(name = "LoginCredientials")
	public static Object[][] credentials() {
	    return new Object[][] { 
	        { "orange", "orangepassword123" }
	    };
	}
	
	@Test(priority = 1, dataProvider = "LoginCredientials")
    public void login(String sUsername, String sPassword) 
    {
		// Find the username and password fields
        WebElement username = driver.findElement(By.id("txtUsername"));
        WebElement password = driver.findElement(By.name("txtPassword"));

        // Enter credentials
        username.sendKeys(sUsername);
        password.sendKeys(sPassword);

        // Click login
        driver.findElement(By.xpath("//*[@id=\"btnLogin\"]")).click();

        System.out.println("Title of the new page is : " + driver.getTitle());
        
        // Read login message
        String loginMessage = driver.findElement(By.id("welcome")).getText();
        
        System.out.println("Confirmation message is - " + loginMessage);
                
        //Verify that the homepage has opened
        message = driver.findElement(By.tagName("h1")).getText();
        System.out.println("You are on " + message);
        
        Assert.assertEquals(message, "Dashboard");
    }
	
	@SuppressWarnings("deprecation")
	@Test(priority = 2)
    public void editUserInformation() throws InterruptedException 
    {
		//Find the “My Info” menu item and click it.
		//driver.findElement(By.xpath("//*[@id=\"menu_pim_viewMyDetails\"]")).click();
		
		WebElement leave = driver.findElement(By.xpath("//*[@id=\"menu_pim_viewMyDetails\"]"));
		
		builder.moveToElement(leave).pause(1000).click(leave).pause(2000).build().perform();
		
		System.out.println("Clicked on My Info tab");
		
		message = driver.findElement(By.xpath("/html/body/div[1]/div[3]/div/div[2]/div[1]/h1")).getText();
		System.out.println("You are on " + message);
        
        //Find the “Edit” menu item and click it.
        driver.findElement(By.xpath("//input[@id='btnSave']")).click();
      	
        //Fill in the Name, Gender, Nationality, and the DOB fields. and Click Save.
        driver.findElement(By.id("personal_txtEmpFirstName")).clear();
        driver.findElement(By.id("personal_txtEmpFirstName")).sendKeys("Archana");
        
        driver.findElement(By.id("personal_txtEmpLastName")).clear();
    	driver.findElement(By.id("personal_txtEmpLastName")).sendKeys("Gidavir");
    	
    	driver.findElement(By.id("personal_optGender_2")).click();
    	
    	driver.findElement(By.id("personal_DOB")).clear();
    	driver.findElement(By.id("personal_DOB")).sendKeys("1990-04-28");
    	
    	driver.findElement(By.xpath("//input[@id='btnSave']")).click();
    	System.out.println("New Information saved successfully..!!");
        
    	Thread.sleep(2000);
    	
    	System.out.println("Updated First Name : " + driver.findElement(By.id("personal_txtEmpFirstName")).getAttribute("value"));
        System.out.println("Updated Last Name : " + driver.findElement(By.id("personal_txtEmpLastName")).getAttribute("value"));
    	System.out.println("Updated Gender : " + driver.findElement(By.id("personal_optGender_2")).isSelected());
    	System.out.println("Updated DOB : " + driver.findElement(By.id("personal_DOB")).getAttribute("value"));
        
    }

	

}

File: Selenium/Project/Activity5.py
from threading import Thread
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver import ActionChains
from selenium.webdriver.support import expected_conditions as ExpectedConditions
from selenium.webdriver.support.wait import WebDriverWait


#create WebDriver instance
with webdriver.Firefox() as driver:

    # Declare the wait variable
    wait = WebDriverWait(driver, timeout=10)

    builder = ActionChains(driver)

    #go to web page
    driver.get("http://alchemy.hguy.co/orangehrm")

    #do the stuff on the page
    pageTitle = driver.title
    print("Title of the page is : " + pageTitle)

#--------------------------------------------------------------------------------------------------
    
    # enter credentials in the username and password fields
    driver.find_element(By.XPATH, "//*[@id=\"txtUsername\"]").send_keys("orange")
	
    driver.find_element(By.XPATH, "//*[@id=\"txtPassword\"]").send_keys("orangepassword123")
	
	#click submit button
    driver.find_element(By.XPATH, "//*[@id=\"btnLogin\"]").click()
    
    # Print the title of the new page
    pageTitle = driver.title
    print("New page title is: ", pageTitle)

    # Read login message
    print("Confirmation message is - " , driver.find_element(By.ID, "welcome").text)

	#Print the confirmation message
    message = driver.find_element(By.TAG_NAME, "h1")
    print("You are on : ", message.text)
    
#--------------------------------------------------------------------------------------------------

    #Find the “My Info” menu item and click it.
    #driver.findElement(By.xpath("//*[@id=\"menu_pim_viewMyDetails\"]")).click()

    leave = driver.find_element(By.XPATH, "//*[@id=\"menu_pim_viewMyDetails\"]")

    builder.move_to_element(leave).pause(5).click(leave).pause(5).perform()

    print("Clicked on My Info tab")

    message = driver.find_element(By.XPATH, "/html/body/div[1]/div[3]/div/div[2]/div[1]/h1").text
    print("You are on " , message)

    #Find the “Edit” menu item and click it.
    driver.find_element(By.XPATH, "//input[@id='btnSave']").click()

    #Fill in the Name, Gender, Nationality, and the DOB fields. and Click Save.
    driver.find_element(By.ID, "personal_txtEmpFirstName").clear()
    driver.find_element(By.ID, "personal_txtEmpFirstName").send_keys("Archana")

    driver.find_element(By.ID, "personal_txtEmpLastName").clear()
    driver.find_element(By.ID, "personal_txtEmpLastName").send_keys("Gidavir")

    driver.find_element(By.ID, "personal_optGender_2").click()

    driver.find_element(By.ID, "personal_DOB").clear()
    driver.find_element(By.ID, "personal_DOB").send_keys("1990-04-28")

    driver.find_element(By.XPATH, "//input[@id='btnSave']").click()
    print("New Information saved successfully..!!")

    # Thread.sleep(20)
    wait.until(ExpectedConditions.visibility_of(driver.find_element(By.ID, "personal_txtEmpFirstName")))

    print("Updated First Name : " , driver.find_element(By.ID, "personal_txtEmpFirstName").get_attribute("value"))
    print("Updated Last Name : " , driver.find_element(By.ID, "personal_txtEmpLastName").get_attribute("value"))
    print("Updated Gender : " , driver.find_element(By.ID, "personal_optGender_2").is_selected())
    print("Updated DOB : " , driver.find_element(By.ID, "personal_DOB").get_attribute("value"))

#--------------------------------------------------------------------------------------------------

    #close the browser
    driver.quit()